shared/teams/team/settings/retention/container.js

```diff
@@ -34,56 +34,64 @@ const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
   let showInheritOption = false
   let showOverrideNotice = false
   let loading = false
-
-  switch (ownProps.entityType) {
+  let entityType = ownProps.entityType
+  switch (entityType) {
     case 'adhoc':
-      if (!ownProps.conversationIDKey) {
-        throw new Error('RetentionPicker needs a conversationIDKey to set adhoc retention policies')
+      if (ownProps.conversationIDKey) {
+        policy = getConversationRetentionPolicy(state, ownProps.conversationIDKey)
+        showInheritOption = false
+        showOverrideNotice = false
+        break
       }
-      policy = getConversationRetentionPolicy(state, ownProps.conversationIDKey)
-      showInheritOption = false
-      showOverrideNotice = false
-      break
+      throw new Error('RetentionPicker needs a conversationIDKey to set adhoc retention policies')
     case 'channel':
-      if (!(ownProps.conversationIDKey && ownProps.teamname)) {
-        throw new Error(
-          'RetentionPicker needs a conversationIDKey AND teamname to set channel retention policies'
-        )
+      if (ownProps.conversationIDKey && ownProps.teamname) {
+        let teamname = ownProps.teamname
+        let conversationIDKey = ownProps.conversationIDKey
+        policy = getConversationRetentionPolicy(state, conversationIDKey)
+        teamPolicy = getTeamRetentionPolicy(state, teamname)
+        loading = !teamPolicy
```

Comment:
why not move this to the bottom so it can be set once

Reply:
there isn't something we can consistently check at the bottom for this flag; in the case of 'adhoc' it's always false, for 'channel', 'big team', and 'small team' it's `!teamPolicy` but by the time we get to the bottom that is assigned to either `policy` or `teamPolicy`. We wouldn't know which to check without adding another switch/conditionals.